test/unit/matrix/test_distribution.cpp

```diff
@@ -0,0 +1,332 @@
+  }
+}
+
+TEST(DistributionTest, ComparisonOperator) {
+  for (const auto& test : tests_constructor) {
+    Distribution obj(test.size, test.block_size, test.grid_size, test.rank, test.src_rank);
+    Distribution obj_eq(test.size, test.block_size, test.grid_size, test.rank, test.src_rank);
+
+    EXPECT_TRUE(obj == obj_eq);
+    EXPECT_FALSE(obj != obj_eq);
+
+    std::vector<Distribution> objs_ne;
+    objs_ne.emplace_back(GlobalElementSize(test.size.rows() + 1, test.size.cols()), test.block_size,
+                         test.grid_size, test.rank, test.src_rank);
+    objs_ne.emplace_back(GlobalElementSize(test.size.rows(), test.size.cols() + 1), test.block_size,
+                         test.grid_size, test.rank, test.src_rank);
+    objs_ne.emplace_back(test.size, TileElementSize(test.block_size.rows() + 1, test.block_size.cols()),
+                         test.grid_size, test.rank, test.src_rank);
+    objs_ne.emplace_back(test.size, TileElementSize(test.block_size.rows(), test.block_size.cols() + 1),
+                         test.grid_size, test.rank, test.src_rank);
+    objs_ne.emplace_back(test.size, test.block_size,
+                         comm::Size2D{test.grid_size.rows() + 1, test.grid_size.cols()}, test.rank,
+                         test.src_rank);
+    objs_ne.emplace_back(test.size, test.block_size,
+                         comm::Size2D{test.grid_size.rows(), test.grid_size.cols() + 1}, test.rank,
+                         test.src_rank);
+    if (test.rank.row() < test.grid_size.rows() - 1) {
+      objs_ne.emplace_back(test.size, test.block_size, test.grid_size,
+                           comm::Index2D(test.rank.row() + 1, test.rank.col()), test.src_rank);
+    }
+    if (test.rank.col() < test.grid_size.cols() - 1) {
+      objs_ne.emplace_back(test.size, test.block_size, test.grid_size,
+                           comm::Index2D(test.rank.row(), test.rank.col() + 1), test.src_rank);
+    }
+    if (test.src_rank.row() < test.grid_size.rows() - 1) {
+      objs_ne.emplace_back(test.size, test.block_size, test.grid_size, test.rank,
+                           comm::Index2D(test.src_rank.row() + 1, test.src_rank.col()));
+    }
+    if (test.src_rank.col() < test.grid_size.cols() - 1) {
+      objs_ne.emplace_back(test.size, test.block_size, test.grid_size, test.rank,
+                           comm::Index2D(test.src_rank.row(), test.src_rank.col() + 1));
+    }
+
+    for (const auto& obj_ne : objs_ne) {
+      EXPECT_TRUE(obj != obj_ne);
+      EXPECT_FALSE(obj == obj_ne);
+    }
+  }
+}
+
+TEST(DistributionTest, CopyConstructor) {
+  for (const auto& test : tests_constructor) {
+    Distribution obj0(test.size, test.block_size, test.grid_size, test.rank, test.src_rank);
+    Distribution obj(test.size, test.block_size, test.grid_size, test.rank, test.src_rank);
+    EXPECT_EQ(obj0, obj);
+
+    Distribution obj_copy(obj);
+    EXPECT_EQ(obj0, obj);
+    EXPECT_EQ(obj, obj_copy);
```

Comment:
Should we check as well `EXPECT_EQ(obj0, obj_copy)`? 
`EXPECT_EQ(obj0, obj)` is already tested above (line 203), right?

Reply:
It is not needed due to transitivity.
Anyway the logic is:
First we check that the object didn't change creating a copy,
then we check that the copy is the same as the original object.